sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -110,11 +111,99 @@ Azure::Response<KeyVaultKey> KeyClient::CreateOctKey(
       Azure::Core::Http::HttpMethod::Post,
       _detail::KeyRequestParameters(octKeyOptions),
       [&keyName](Azure::Core::Http::RawResponse const& rawResponse) {
-        return _detail::KeyVaultKeyDeserialize(keyName, rawResponse);
+        return _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(keyName, rawResponse);
       },
       {_detail::KeysPath, keyName, "create"});
 }
 
+Azure::Response<KeyPropertiesSinglePage> KeyClient::GetPropertiesOfKeysSinglePage(
+    GetPropertiesOfKeysSinglePageOptions const& options,
+    Azure::Core::Context const& context) const
+{
+  if (!options.ContinuationToken) // First page when no continuation token //
```

Comment:
This code can be simplified to reduce the number of repeated methods.
Make the query an empty string to start, add to it if you have a continuationToken and again if MaxResults is set.  Pass the query always since empty query is a no-op

Reply:
the continuationToken is a full `URL`,  I might not understand what you are proposing here